### PR TITLE
Ccp 81

### PR DESCRIPTION
**Pull request by mkarthik90**
[CCP-81] Made changes to create wordnet wrapper and concept search.

### CODE REVIEW

**Comment by jdamerow** [comment]
please resolve conflicts

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/wrapper/ConceptEntryWrapper.java [hunk 1]
```diff
@@ -17,6 +19,8 @@
  * @author Julia Damerow
  * 
  */
+@Component("conceptEntryWrapper")
+@Scope(value = "prototype")
 public class ConceptEntryWrapper implements Serializable {
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/wrapper/ConceptEntryWrapper.java]: Didn't we discuss this? That should be just a simple class, no scope, no component. And then the uri is a property that is set in the wrapper creator.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/core/ConceptEntry.java [hunk 2]
```diff
@@ -330,6 +332,14 @@ public boolean equals(Object obj) {
...
 	}
 
+    public String getUri() {
+        return uri;
+    }
+
+    public void setUri(String uri) {
+        this.uri = uri;
+    }
+
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/core/ConceptEntry.java]: I don't think you need this property here. Where do you use it?

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/wrapper/ConceptEntryWrapper.java (hunk 1 above)
Comment (mkarthik90) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/wrapper/ConceptEntryWrapper.java]: @jdamerow  Yes Julia, without having conceptentrywrapper as a spring bean we cannot autowire the bean in ConceptWrapperAddController search method(). If that is fine I will remove the @Component and create the object through new keyword.

**Comment by jdamerow** [comment]
what about the other comment?

**Comment by mkarthik90** [comment]
@jdamerow

**Comment by jdamerow** [comment]
can we put that in the wrapper? we don't want to store the URI in the database where it would end up if it's part of the conceptentry.

**Comment by mkarthik90** [comment]
@jdamerow  I have made the changes for URI and verified

**Comment by jdamerow** [comment]
please resolve conflicts

**Comment by mkarthik90** [comment]
@jdamerow  Resolved merge conflicts. Could you please verify and merge branch

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/wrapper/ConceptEntryWrapper.java (hunk 1 above)
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/wrapper/ConceptEntryWrapper.java]: I was so sure I responded to that. Hm, well... apparently not.

Yes, remove component and create a new object with new.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/wrapper/impl/ConceptEntryWrapperCreator.java [hunk 3]
```diff
@@ -31,7 +35,13 @@
 
 	@Autowired
 	private IUserManager usersManager;
+	
+	@Autowired
+    private IURIHelper helper;
+	
+	private BeanFactory beanFactory;
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/wrapper/impl/ConceptEntryWrapperCreator.java]: I don't think you still need that, do you?

**Pull request closed without merging**